### ISSUE TITLE
Wrong function name or function implementation (env_create_mutex behave as semaphore)

### ISSUE THREAD

**Issue by Hadatko** [NXPmicro/rpmsg-lite#15]
env_create_mutex for freertos has semaphore implementation: https://www.geeksforgeeks.org/mutex-vs-semaphore/
This can consufe users during new ports or debugging.

**Comment by MichalPrincNXP** [comment]
@Hadatko , you are right, it would be better to use the xSemaphoreCreateMutex API to be accurate. I guess the xSemaphoreCreateCounting usage has been chosen because of the smaller footprint at the time this project has been created.
